### PR TITLE
Add id token support

### PR DESCRIPTION
**Pull request by stanleyphu**
This PR adds support to the authentication middleware to support id tokens returned by authentication providers. Currently, this only adds support for RSA encrypted tokens by requiring a public key in the configuration which is used by the middleware to verify the token. 

By default, all endpoints do not accept token authentication unless specified by the `EnableSessionAuth` flag. However, the authorize endpoint has been updated to enable session auth, allowing checks to be made without requiring a subject in each warrant. In this use case, the authorize endpoint will automatically append all warrants to use the user id associated with the token.

### CODE REVIEW

File: pkg/authz/check/handlers.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package authz
 import (
 	"net/http"
 
+	objecttype "github.com/warrant-dev/warrant/pkg/authz/objecttype"
+	authz "github.com/warrant-dev/warrant/pkg/authz/warrant"
```
Comment (kkajla12) [inline pkg/authz/check/handlers.go:7]: We should alias this import as `warrant`:
```go
warrant "github.com/warrant-dev/warrant/pkg/authz/warrant"
```

File: pkg/authz/warrant/spec.go [hunk 2]
```diff
@@ -184,3 +184,13 @@ func StringToWarrantSpec(warrantString string) (*WarrantSpec, error) {
...
 	}, nil
 }
+
+// SessionWarrantSpec type
+type SessionWarrantSpec struct {
+	ObjectType string                 `json:"objectType" validate:"required,valid_object_type"`
+	ObjectId   string                 `json:"objectId" validate:"required,valid_object_id"`
+	Relation   string                 `json:"relation" validate:"required,valid_relation"`
+	Context    context.ContextSetSpec `json:"context,omitempty"`
+	IsImplicit *bool                  `json:"isImplicit,omitempty"`
```
Comment (kkajla12) [inline pkg/authz/warrant/spec.go:194]: I think we can remove `IsImplicit`

**Review by kkajla12 — CHANGES_REQUESTED** [review]

File: pkg/config/config.go [hunk 3]
```diff
@@ -63,9 +63,10 @@ type EventstoreConfig struct {
 }
 
 type AuthConfig struct {
-	Provider       string `mapstructure:"provider"`
-	PublicKey      string `mapstructure:"publicKey"`
-	UserIdentifier string `mapstructure:"userIdentifier"`
+	Provider         string `mapstructure:"provider"`
+	PublicKey        string `mapstructure:"publicKey"`
+	UserIdentifier   string `mapstructure:"userIdentifier"`
```
Comment (kkajla12) [inline pkg/config/config.go:68]: Maybe we can rename these two properties to `UserIdClaim` and `TenantIdClaim` or `UserClaim` and `TenantClaim`

Comment (stanleyphu) [reply pkg/config/config.go]: Updated to `UserIdClaim` and `TenantIdClaim`

Comment (kkajla12) [reply pkg/config/config.go]: Need to update the sample configs in the mysql and postgres READMEs as well

**Review by kkajla12 — APPROVED** [review]

**Pull request merged**